src/System.Management.Automation/engine/remoting/server/ServerRunspacePoolDriver.cs

```diff
@@ -1889,87 +1889,87 @@ public override bool InBreakpoint
             _wrappedDebugger.Value.SetBreakpoints(breakpoints, runspaceId);
 
         /// <summary>
         /// Get a breakpoint by id, primarily for Enable/Disable/Remove-PSBreakpoint cmdlets.
         /// </summary>
         /// <param name="id">Id of the breakpoint you want.</param>
-        /// <param name="runspaceId">The runspace id of the runspace you want to interact with. Defaults to null (current runspace).</param>
+        /// <param name="runspaceId">The runspace id of the runspace you want to interact with. A null value will use the current runspace.</param>
         /// <returns>The breakpoint with the specified id.</returns>
-        public override Breakpoint GetBreakpoint(int id, int? runspaceId = null) =>
+        public override Breakpoint GetBreakpoint(int id, int? runspaceId) =>
             _wrappedDebugger.Value.GetBreakpoint(id, runspaceId);
 
         /// <summary>
         /// Returns breakpoints on a runspace.
         /// </summary>
-        /// <param name="runspaceId">The runspace id of the runspace you want to interact with. Defaults to null (current runspace).</param>
+        /// <param name="runspaceId">The runspace id of the runspace you want to interact with. A null value will use the current runspace.</param>
         /// <returns>A list of breakpoints in a runspace.</returns>
-        public override List<Breakpoint> GetBreakpoints(int? runspaceId = null) =>
+        public override List<Breakpoint> GetBreakpoints(int? runspaceId) =>
             _wrappedDebugger.Value.GetBreakpoints(runspaceId);
 
         /// <summary>
         /// Sets a command breakpoint in the debugger.
         /// </summary>
-        /// <param name="command">The name of the command that will trigger the breakpoint. This value is required and may not be null.</param>
+        /// <param name="command">The name of the command that will trigger the breakpoint. This value may not be null.</param>
         /// <param name="action">The action to take when the breakpoint is hit. If null, PowerShell will break into the debugger when the breakpoint is hit.</param>
         /// <param name="path">The path to the script file where the breakpoint may be hit. If null, the breakpoint may be hit anywhere the command is invoked.</param>
-        /// <param name="runspaceId">The runspace id of the runspace you want to interact with. Defaults to null (current runspace).</param>
+        /// <param name="runspaceId">The runspace id of the runspace you want to interact with. A null value will use the current runspace.</param>
         /// <returns>The command breakpoint that was set.</returns>
-        public override CommandBreakpoint SetCommandBreakpoint(string command, ScriptBlock action = null, string path = null, int? runspaceId = null) =>
+        public override CommandBreakpoint SetCommandBreakpoint(string command, ScriptBlock action, string path, int? runspaceId) =>
```

Comment:
Shouldn't there be overloads for this? It looks like you just removed the default parameters but this PR is suppose to be a non-functional change.

Reply:
I discussed it with @daxian-dbw and the breakdown is:

- Providing overloads for each default parameter would mean 6 or 24 overloads in some cases
- Although default parameters make sense for a PowerShell cmdlet, this is an API, so the UX is different; pith is not as valuable and may not actually be useful anyway
- In particular I noticed that
  - in the Set-PSBreakpoint case, the defaults were only used once and in that case it seemed to me that setting `path: null` actually seemed better
  - in PSES, we use reflection to call the methods, so we don't benefit at all from the default parameters
  - the inheriting classes also needed to set default parameters. Now they just implement the main virtual method
- Given the unusualness of needing to target a different runspace, and the way I'd already overloaded the other APIs, I kept the `runspaceId` overloads
- Each parameter is documented (I updated the documentation) so that API consumers will know if they should set parameters as `null`

Reply:
I'm fine with this but it seems like you're now breaking the API (which I understand I already broke) for the sake of consistency?

That's not typically the approach we've taken to breaking changes.

> Providing overloads for each default parameter would mean 6 or 24 overloads in some cases

This is one of the reasons why the addition of the `runspaceId` with a default parameter was done and since there were already default parameters, it fit right in - despite it being a breaking change.

Reply:
> but it seems like you're now breaking the API

Yeah that's true in the sense that I'm changing the API, but it was only introduced in a 7 preview; from the perspective of previously GA's PowerShells (6.2, 5.1) this PR restores one API and doesn't break any others.

> for the sake of consistency?

An inconsistent API would be ok, but the default parameters [can lead to runtime exceptions in consuming code if we ever add more APIs](https://haacked.com/archive/2010/08/10/versioning-issues-with-optional-arguments.aspx/). If we ever decided we needed to provide more overloads or parameters for these breakpoint APIs, we'd need to tread very carefully if we keep the default parameters.

I don't really like the extent of these changes, but they try to set API to something we can maintain and extend before 7 is locked in place.

Reply:
> this PR restores one API and doesn't break any others

you're making parameters that use to have a default value to not having a default value. That's breaking the previous GA API, no?

Reply:
> you're making parameters that use to have a default value to not having a default value

Yes, absolutely.

> That's breaking the previous GA API, no?

Not quite, because that API was introduced in https://github.com/PowerShell/PowerShell/commit/cc0fed479a3e455b746a3d12597f078462f2d644, released in 7.0.0-preview.4. So the API's never GA'd.

This change is like the `-loadprofile` -> `-login` change, where it was released in a preview version but never GA'd.

Reply:
This PR is marked with `GA-Consider` milestone, and once approved will be included in GA release.

Reply:
Ah so you've only added the overloads to the previously GA'd APIs. That was the detail I was missing.

I'm ok with this then.

Reply:
@TylerLeonhardt is this resolved?

Reply:
Yes it is